File: GC.WebSpace/ClientApp/domain/records/ads/adType.ts
import { Never } from "../../../tools/never";

export enum AdType {
    SelfWritten = 1,
    Offered = 2
}

export namespace AdType {
    export function getDisplayName(type: AdType) {
        switch (type) {
            case AdType.SelfWritten: return 'Самописное';
            case AdType.Offered: return 'Предложенное';
            default: throw new Never(type);
        }
    }

    export function getAll(): AdType[] {
        return Object.keys(AdType)
            .filter(v => isNaN(Number(v)) === false)
            .map(key => Number(key) as AdType);
    }
}

File: GC.Domain/Records/Ads/AdType.cs
using System;

namespace GC.Domain.Records.Ads
{
    public enum AdType
    {
        SelfWritten = 1,
        Offered = 2
    }

    public static class AdTypeExtensions
    {
        public static String GetDisplayName(this AdType type)
        {
            switch(type)
            {
                case AdType.SelfWritten: return "Самописное объявление";
                case AdType.Offered: return "Предложенное объявление";
                default: throw new Exception("Не указана расшифровка для типа объявления");
            }
        }
    }
}
